File: backend/mongddang/src/main/java/com/onetwo/mongddang/domain/vital/dto/BloodGlucoseRequest.java
package com.onetwo.mongddang.domain.vital.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import com.onetwo.mongddang.domain.vital.model.BloodGlucose;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.validation.constraints.Null;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BloodGlucoseRequest {

    private Integer bloodSugarLevel;

    private LocalDateTime measurementTime;

    private String packageName;

}

File: frontend/android/app/src/main/java/com/mongddang/app/data/local/entity/BloodGlucoseRequest.kt
package com.mongddang.app.data.local.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class BloodGlucoseRequest(
    @SerialName("bloodSugarLevel")
    val bloodSugarLevel: Int = 0,
    @SerialName("measurementTime")
    val measurementTime: LocalDateTime? = null,
    @SerialName("status")
    val status: Status? = Status.isStatus(bloodSugarLevel),
    @SerialName("notification")
    val notification: Boolean? = false,
    @SerialName("packageName")
    val packageName: String? = null
)
